src/fknni/faiss/faiss.py

```diff
@@ -32,87 +31,172 @@ def __init__(
+                continue
+
+            # Extract the features from X that was used to train FAISS, and compute the sparseness matrix
+            x_imputed = self.X_full[::, training_indices]
+            x_imputed_missing_mask = np.isnan(x_imputed)
+
+            # Iterate through the data to impute, ignoring already imputed rows
+            sample_missing_mask = []
+            for idx in np.where(x_imputed_missing_mask.any(axis=1))[0]:
+                # Extract the sample
+                sample_missing_mask = x_imputed_missing_mask[idx]
+                sample = x_imputed[idx]
+
+                # We need to prefill the query vector as FAISS doesn't accept nan´s
+                sample[sample_missing_mask] = global_fallbacks_[training_indices][sample_missing_mask]
+                missing_cols = np.where(sample_missing_mask)[0]
+
+                # Call FAISS and retrieve data
+                distances, indices = index.search(sample.reshape(1, -1), self.n_neighbors)
+                assert len(indices[0]) == self.n_neighbors
+                valid_indices = indices[0][indices[0] >= 0] # Filter out negative indices because they are FAISS error codes
+
+                # FAISS couldn't find any neighbor, use fallback values, and go to next row
+                if len(valid_indices) == 0:
+                    self._warn_fallback()
+                    x_imputed[idx, missing_cols] = sample[missing_cols]
+                    continue
+
+                # FAISS couldn't find the amount of requested neighbors, warn user and proceed
+                if len(valid_indices) < self.n_neighbors:
+                    if not self.warned_unsufficient_neighbors:
+                        logger.warning(f"FAISS couldn't find all the requested neighbors at least once")
+                        self.warned_unsufficient_neighbors = True
+
+                # Apply strategy on neighbors data
+                neighbors = training_data[valid_indices]
+                if self.strategy == "weighted":
+                    weights = 1 / (distances[0] + 1e-10)[:, np.newaxis]
+                    neighbor_vals = neighbors[:, missing_cols]
+                    weighted_sum = np.nansum(neighbor_vals * weights, axis=0)
+                    weight_sum = np.nansum(weights, axis=0)
+                    x_imputed[idx, missing_cols] = weighted_sum / weight_sum
+                else:
+                    func = np.nanmean if self.strategy == "mean" else np.nanmedian
+                    x_imputed[idx, missing_cols] = func(neighbors[:, missing_cols], axis=0)
+
+            # Transfer back to X
+            # Features added by _fit_train_imputer for training purpose only are placed on the right
+            # so we can just select the features on the left
+            self.X_full[::, feature_indices_being_imputed] = x_imputed[::, np.arange(len(feature_indices_being_imputed))]
```

Comment:
```suggestion
            self.X_full[:, feature_indices_being_imputed] = x_imputed[:, np.arange(len(feature_indices_being_imputed))]
```

also here, we don't need the double `::` right?